src/views/Home.jsx

```diff
@@ -1,12 +1,58 @@
 import './Home.css';
+import { useCallback, useState } from 'react';
+import { checkListToken } from '../api/firebase';
+import { useNavigate } from 'react-router-dom';
+
+export function Home({ handleNewToken, setListToken, listToken }) {
+	const [token, setToken] = useState('');
+	const [listNotFound, setlistNotFound] = useState('');
+	const redirect = useNavigate();
+
+	if (listToken) {
+		redirect('/list');
+	}
+
+	const handleClick = useCallback(() => {
+		handleNewToken();
+	}, [handleNewToken]);
+
+	const handleFormChange = (e) => {
+		const value = e.target.value;
+		setToken(value);
+	};
+
+	const handleTokenSubmit = async (e) => {
+		e.preventDefault();
+
+		const checkTokenExists = await checkListToken(token);
+
+		if (checkTokenExists) {
+			setListToken(token);
+			redirect('/list');
+		} else {
+			setlistNotFound('Could not find this list');
+		}
+	};
 
-export function Home({ handleNewToken }) {
 	return (
 		<div className="Home">
 			<p>
 				Hello from the home (<code>/</code>) page.
 			</p>
-			<button onClick={handleNewToken}>Create a new list!</button>
+			<button onClick={handleClick}>Create a new list!</button>
+			<p> Want to join an existing list? </p>
+			<form onSubmit={handleTokenSubmit}>
+				<label htmlFor="token"> Enter Token:</label>
+				<input
+					type="text"
+					name="token"
+					id="token"
+					value={token}
```

Comment:
Edge case...do we need to consider case sensitivity? ie token is 'my test list'....user enters 'My Test List'
or is the assumption that the user knows that it must match exactly, similar to a password. Could use the toLowerCase() method if its something we decide to consider.

Reply:
@ehammes theoretically both "my test list" and "My test list" could be two different lists in the database and both exist

Reply:
Future improvement...may want to account for form validation here if the user clicks 'Join' without an input value

Reply:
Always a great idea to validate your input. Checkout the built-in [validations](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#using_built-in_form_validation) as a first line of defense. You can then read up on using javascript for input validation, but at the very least normalizing your strings in someway will probably be a good idea.

Reply:
Copy! Added in a required form validation and accounted for token case insensitivity